### DISCUSSION TITLE
Why Identity Syetem is not recommened for API apps?

### DISCUSSION THREAD

**Discussion by xAdamQ** [dotnet/aspnetcore#27641] [Q&A]
I want to make a Simple SignalR game but I am struggling with authentication for a while know. I can't understand the differences between IS4 and Identity System.
But Microsoft documentation says

> ASP.NET Core Identity adds user interface (UI) login functionality to ASP.NET Core web apps. To secure web APIs and SPAs, use one of the following:
    Azure Active Directory
    Azure Active Directory B2C (Azure AD B2C)
    IdentityServer4

My Client is not a browser and I don't even need user interface for login, the authentication happens with a token from 3rd party.
I saw some tutorials uses Identity System in API projects but I don't know if this right or not. 
The main reason I don't want to use IS4 it's becoming commercial.

I may have some misunderstanding and I hope some one gives me a detailed answer.

**Comment by blowdart** [comment]
Detailed is hard, because as you've discovered it is very context specific.

Your case is you don't do logins at all, so don't use anything at all, just validate the token. How you do that depends on what the token is, and how you send it. If it's a JWT, then you use the JWT middleware, and so on.

The reason we recommend Identity Server, or AAD for APIs is they use a standard mechanism than pretty much every UI framework has plugins for, and they avoid having a user's application do logins or store credentials.
